### PR TITLE
Implement recursive fs watch on Windows

### PR DESCRIPTION
**Pull request by bpasero**
This pull request contains the changes from https://github.com/joyent/libuv/pull/1473 (@ghostoy fyi). The only change I did was to use the newly introduced UV_FS_EVENT_RECURSIVE flag instead of the custom added one to not break the ABI. This was the complaint in the older pull request and the reason why it was not accepted.

The included test was also taken from the original pull request and still passes for me. Some testing on my Windows 10 box also shows that events are flowing. I did my testing in io.js with my change included in libuv.

### CODE REVIEW

File: src/win/fs-event.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ static void uv_fs_event_queue_readdirchanges(uv_loop_t* loop,
   if (!ReadDirectoryChangesW(handle->dir_handle,
                              handle->buffer,
                              uv_directory_watcher_buffer_size,
-                             FALSE,
+							               (handle->flags & UV_FS_EVENT_RECURSIVE) ? TRUE : FALSE,
```
Comment (saghul) [inline src/win/fs-event.c]: I don't think this is going to fly. UV_FS_EVENT_RECURSIVE has the same value as UV_HANDLE_ENDGAME_QUEUED, which is used internally by Windows handles. That's why the initial PR added a new field.

**Comment by ghostoy** [comment]
+1. Any update?

**Comment by saghul** [comment]
@ghostoy look at my note. The current approach adds a flag that looks like it would clash. Adding a new field is not an option since we'd break the ABI.

**Comment by piscisaureus** [comment]
Generally I'm down with the idea, and the patch looks okay to me too.

Could you try to adhere to the code style used in libuv? At the very least don't mix spaces and tabs.

**Comment by saghul** [comment]
@piscisaureus is this ok? https://github.com/libuv/libuv/pull/198#discussion_r24483338

**Comment by piscisaureus** [comment]
> I don't think this is going to fly. UV_FS_EVENT_RECURSIVE has the same value as UV_HANDLE_ENDGAME_QUEUED, which is used internally by Windows handles. That's why the initial PR added a new field.

Yeah, good catch - that has to be fixed too. (should not be too hard, just make sure the values don't overlap.)

**Comment by ReklatsMasters** [comment]
Any problem? Change  [UV_HANDLE_ENDGAME_QUEUED](https://github.com/libuv/libuv/blob/v1.x/src/win/internal.h#L67) to 
```c
#define UV_HANDLE_ENDGAME_QUEUED                0x00000008
```
It\`s used only in 2 files - `src\win\handle-inl.h` and `src\win\internal.h`.
@bpasero

**Comment by saghul** [comment]
Superseded by https://github.com/libuv/libuv/pull/421

**Pull request closed without merging**